spec/decision_spec.rb

```diff
@@ -1,93 +1,201 @@
 require "spec_helper"
+require "chef-config/windows"
+require "chef/telemetry/decision"
+require "tmpdir"
+require "logger"
+
+# This should probably be in the CI script, not here
+ENV["CI_TELEMETRY_PROMPT_TIMEOUT"] = "1"
 
 RSpec.describe Chef::Telemetry::Decision do
-  let(:env) { {} }
-  let(:env_pwd) { "/path/to/cwd" }
-  let(:home) { "/Users/chef_user" }
+  it "has a version number" do
+    expect(Chef::Telemetry::VERSION).not_to be nil
+  end
 
-  before do
-    stub_const("ENV", env)
+  let(:logger) { l = Logger.new(STDERR); l.level = Logger::ERROR; l }
+  let(:output) do
+    buffer = StringIO.new
+    allow(buffer).to receive(:isatty).and_return(true)
+    buffer
+  end
+  let(:input) { STDIN }
+  let(:opts) { { logger: logger, output: output, input: input } }
+  let(:dec) { Chef::Telemetry::Decision.new(opts) }
 
-    if ChefConfig.windows?
-      env["CD"] = env_pwd
-    else
-      env["PWD"] = env_pwd
-    end
+  def check_option_behavior(test_params = { should_be_enabled: false, should_persist: false, should_prompt: false } )
+    Dir.mktmpdir do |dir|
 
-    allow(ChefConfig::PathHelper).to receive(:home).with(".chef").and_return(File.join(home, ".chef"))
-  end
+      dec.file_decision.local_dir = dir
```

Comment:
This should pass without setting `local_dir` explicitly, since check_and_persist will do do that - maybe better to test that after `check_and_persist` is done, `local_dir` is set to `dir`?

Reply:
Yep, this is cruft - thanks!